subsys/net/ip/dhcpv4.c

```diff
@@ -203,6 +203,9 @@ static struct net_pkt *dhcpv4_create_message(struct net_if *iface, uint8_t type,
 
 	pkt = net_pkt_alloc_with_buffer(iface, size, AF_INET,
 					IPPROTO_UDP, K_FOREVER);
```

Comment:
Considering `K_FOREVER` timeout, doesn't it wait forever until there is enough space to allocate the pkt?